File: Server/src/main/java/com/api/server/foodb/FoodcomexCompound.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "foodcomex_compounds")
public class FoodcomexCompound {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "compound_id", nullable = false)
    private Integer compoundId;

    @Size(max = 255)
    @Column(name = "origin")
    private String origin;

    @Size(max = 255)
    @Column(name = "storage_form")
    private String storageForm;

    @Size(max = 255)
    @Column(name = "maximum_quantity")
    private String maximumQuantity;

    @Size(max = 255)
    @Column(name = "storage_condition")
    private String storageCondition;

    @Size(max = 255)
    @Column(name = "contact_name")
    private String contactName;

    @Lob
    @Column(name = "contact_address")
    private String contactAddress;

    @Size(max = 255)
    @Column(name = "contact_email")
    private String contactEmail;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "export")
    private Boolean export;

    @Lob
    @Column(name = "purity")
    private String purity;

    @Lob
    @Column(name = "description")
    private String description;

    @Size(max = 255)
    @Column(name = "spectra_details")
    private String spectraDetails;

    @Size(max = 255)
    @Column(name = "delivery_time")
    private String deliveryTime;

    @Size(max = 255)
    @Column(name = "stability")
    private String stability;

    @Column(name = "admin_user_id")
    private Integer adminUserId;

    @Size(max = 255)
    @NotNull
    @Column(name = "public_id", nullable = false)
    private String publicId;

    @Size(max = 255)
    @Column(name = "cas_number")
    private String casNumber;

    @Size(max = 255)
    @Column(name = "taxonomy_class")
    private String taxonomyClass;

    @Size(max = 255)
    @Column(name = "taxonomy_family")
    private String taxonomyFamily;

    @Size(max = 255)
    @Column(name = "experimental_logp")
    private String experimentalLogp;

    @Size(max = 255)
    @Column(name = "experimental_solubility")
    private String experimentalSolubility;

    @Size(max = 255)
    @Column(name = "melting_point")
    private String meltingPoint;

    @Size(max = 255)
    @Column(name = "food_of_origin")
    private String foodOfOrigin;

    @Lob
    @Column(name = "production_method_reference_text")
    private String productionMethodReferenceText;

    @Size(max = 255)
    @Column(name = "production_method_reference_file_name")
    private String productionMethodReferenceFileName;

    @Size(max = 255)
    @Column(name = "production_method_reference_content_type")
    private String productionMethodReferenceContentType;

    @Column(name = "production_method_reference_file_size")
    private Integer productionMethodReferenceFileSize;

    @Column(name = "production_method_reference_updated_at")
    private Instant productionMethodReferenceUpdatedAt;

    @Size(max = 255)
    @Column(name = "elemental_formula")
    private String elementalFormula;

    @Size(max = 255)
    @Column(name = "minimum_quantity")
    private String minimumQuantity;

    @Size(max = 255)
    @Column(name = "quantity_units")
    private String quantityUnits;

    @Lob
    @Column(name = "available_spectra")
    private String availableSpectra;

    @Lob
    @Column(name = "storage_conditions")
    private String storageConditions;

}

File: DotNetServer/DotNetServer/Entities/Foodb/FoodcomexCompound.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Entities.Foodb;

[Table("foodcomex_compounds")]
[Index("AdminUserId", Name = "index_foodcomex_compounds_on_admin_user_id")]
[Index("CompoundId", Name = "index_foodcomex_compounds_on_compound_id")]
public partial class FoodcomexCompound
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("compound_id")] public int CompoundId { get; set; }

    [Column("origin")] [StringLength(255)] public string? Origin { get; set; }

    [Column("storage_form")]
    [StringLength(255)]
    public string? StorageForm { get; set; }

    [Column("maximum_quantity")]
    [StringLength(255)]
    public string? MaximumQuantity { get; set; }

    [Column("storage_condition")]
    [StringLength(255)]
    public string? StorageCondition { get; set; }

    [Column("contact_name")]
    [StringLength(255)]
    public string? ContactName { get; set; }

    [Column("contact_address", TypeName = "text")]
    public string? ContactAddress { get; set; }

    [Column("contact_email")]
    [StringLength(255)]
    public string? ContactEmail { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [Column("export")] public bool? Export { get; set; }

    [Column("purity", TypeName = "text")] public string? Purity { get; set; }

    [Column("description", TypeName = "text")]
    public string? Description { get; set; }

    [Column("spectra_details")]
    [StringLength(255)]
    public string? SpectraDetails { get; set; }

    [Column("delivery_time")]
    [StringLength(255)]
    public string? DeliveryTime { get; set; }

    [Column("stability")]
    [StringLength(255)]
    public string? Stability { get; set; }

    [Column("admin_user_id")] public int? AdminUserId { get; set; }

    [Column("public_id")]
    [StringLength(255)]
    public string PublicId { get; set; } = null!;

    [Column("cas_number")]
    [StringLength(255)]
    public string? CasNumber { get; set; }

    [Column("taxonomy_class")]
    [StringLength(255)]
    public string? TaxonomyClass { get; set; }

    [Column("taxonomy_family")]
    [StringLength(255)]
    public string? TaxonomyFamily { get; set; }

    [Column("experimental_logp")]
    [StringLength(255)]
    public string? ExperimentalLogp { get; set; }

    [Column("experimental_solubility")]
    [StringLength(255)]
    public string? ExperimentalSolubility { get; set; }

    [Column("melting_point")]
    [StringLength(255)]
    public string? MeltingPoint { get; set; }

    [Column("food_of_origin")]
    [StringLength(255)]
    public string? FoodOfOrigin { get; set; }

    [Column("production_method_reference_text", TypeName = "text")]
    public string? ProductionMethodReferenceText { get; set; }

    [Column("production_method_reference_file_name")]
    [StringLength(255)]
    public string? ProductionMethodReferenceFileName { get; set; }

    [Column("production_method_reference_content_type")]
    [StringLength(255)]
    public string? ProductionMethodReferenceContentType { get; set; }

    [Column("production_method_reference_file_size")]
    public int? ProductionMethodReferenceFileSize { get; set; }

    [Column("production_method_reference_updated_at", TypeName = "datetime")]
    public DateTime? ProductionMethodReferenceUpdatedAt { get; set; }

    [Column("elemental_formula")]
    [StringLength(255)]
    public string? ElementalFormula { get; set; }

    [Column("minimum_quantity")]
    [StringLength(255)]
    public string? MinimumQuantity { get; set; }

    [Column("quantity_units")]
    [StringLength(255)]
    public string? QuantityUnits { get; set; }

    [Column("available_spectra", TypeName = "text")]
    public string? AvailableSpectra { get; set; }

    [Column("storage_conditions", TypeName = "text")]
    public string? StorageConditions { get; set; }
}